### PR TITLE
ENH: ``default_rng`` coerces ``RandomState`` to ``Generator``

### PR DESCRIPTION
**Pull request by andyfaff**
This PR proposes to allow `default_rng` to coerce `RandomState` to a `Generator`.

This is useful when downstream projects are doing a [SPEC7]( https://github.com/andyfaff/numpy/pull/new/random) transition from legacy `RandomState` to `Generator`.

One illustration is `RandomState` and `Generator` offer slightly different methods. e.g. `RandomState.randint`/`Generator.integers`, `RandomState.random_sample`/`Generator.random`, etc.

Codebases have therefore got to be written to cope with either class. If `default_rng` can do the normalisation then code only has to be written to deal with `Generator`.
The downstream libraries need to understand how a coerced object may be different.

e.g.
`RandomState.random_sample` and `Generator.random` (where the `Generator` was coerced from the same `RandomState` object) can give an identical stream of floats. However, the same can't be said for generating integers.

### CODE REVIEW

**Comment by rkern** [comment]
No copying of state, please. Just grab the `._bit_generator` attribute from the `RandomState` so the `Generator` uses the same instance.

**Comment by andyfaff** [comment]
@rkern, continuing the conversation from https://github.com/scipy/scipy/pull/21888#issuecomment-2477800352.

Is this the correct way to do it? I couldn't see another way of directly accessing the BitGenerator.

File: numpy/random/tests/test_direct.py [hunk 1]
```diff
@@ -559,3 +559,22 @@ def test_passthrough(self):
...
+        assert isinstance(rg.bit_generator, MT19937)
+        assert rg.bit_generator is rs._bit_generator
+
+        # RandomState with a non MT19937 bit generator
+        _original = np.random.get_bit_generator()
+        bg = PCG64(12342298)
+        np.random.set_bit_generator(bg)
+        rs = np.random.mtrand._rand
+        rg = default_rng(rs)
+        assert rg.bit_generator is bg
```
Comment (rkern) [inline numpy/random/tests/test_direct.py:576]: Ignorable nit: I'd probably just delete this section or construct a fresh `RandomState` instance with the specified `bg` if you really want to test a non-MT19937 `BitGenerator` (e.g. `rs = RandomState(bg)`). Mucking with the global `RandomState` instance isn't testing anything about `default_rng()`.

**Review by rkern — APPROVED** [review]
LGTM! I don't think we need extra documentation in the narrative docs, but if we do, we can do that later. Thank you!

**Comment by charris** [comment]
Thanks @andyfaff .

**Pull request merged**